api/schema_system.go

```diff
@@ -117,3 +158,24 @@ func UnmarshalNamingRulePayloadFormat(payload string) (*regexp.Regexp, error) {
 	}
 	return format, nil
 }
+
+// UnmarshalNamingRulePayloadAsTemplate will unmarshal payload to NamingRulePayload and extract all the template keys.
```

Comment:
consider add a unit test for this.

Reply:
Continue working on this branch for index naming advisor. Will close this PR and re-open another one later.